ResearchKitTests/ORKRecorderTests.m

```diff
@@ -530,7 +532,7 @@ - (void)testPedometerRecorder {
 - (void)testTouchRecorder {
     
     Class recorderClass = [ORKTouchRecorder class];
-    ORKTouchRecorder *recorder = (ORKTouchRecorder *)[self createRecorder:[ORKTouchRecorderConfiguration new]];
+    ORKTouchRecorder *recorder = (ORKTouchRecorder *)[self createRecorder:[[ORKTouchRecorderConfiguration alloc] initWithIdentifier:@"touch"]];
     
     XCTAssertTrue([recorder isKindOfClass:recorderClass], @"");
     
```

Comment:
Also, add a check in `checkResult` to verify the result's identifier is correct.